crates/astria-conductor/src/conductor/inner.rs

```diff
@@ -215,18 +212,29 @@ impl Conductor {
     /// because kubernetes issues SIGKILL 30 seconds after SIGTERM, giving 5 seconds
     /// to abort the remaining tasks.
     #[instrument(skip_all)]
-    async fn shutdown(mut self) {
-        self.shutdown.cancel();
+    async fn shutdown(
+        mut self,
+        exit_reason: eyre::Result<&str>,
```

Comment:
If you pass in the latest failed task here with it's `&'static str` name and and its shutdown reason/error, then you can test this immediately instead of needing to collect all the shutdown reasons and parsing them.